bond/commands/upgrade.py

```diff
@@ -91,6 +92,9 @@ class UpgradeCommand(BaseCommand):
         "--target": {
             "help": "override detected target. Useful in development, but may cause irreversible device malfunction!"
         },
+        "--rollback": {
+            "help": "number of versions to roll back. 0 is the current version, 1 is the one before that and so on."
```

Comment:
I don't like the name `rollback`. It makes it seem like it will undo a previous update. Like, that `--rollback 2` repeated twice will get you back 4 versions behind latest.

Perhaps `--oldversion`? (A nod to the venerable www.oldversion.com).

Also, this would be the first time (other than forensics) that we will look back into the history of the versions JSON. I had been thinking of pushing a new firmware as essentially making the old firmware inaccessible. There could be broken versions in the history which, after this change, we would need to prune. --- Something we should be doing anyways.

An interesting (but undesirable) possibility is allowing a rollback by switching boot partition without re-downloading. That would exactly rollback to the previous version. But that would require firmware changes for little benefit.

Another possibility would be `bond reset rescue` which could go back to the rescue firmware. That is definitely not recommended though, as the database is wiped and it may be difficult (though we commit to it never being impossible) to upgrade from an increasingly ancient rescue image.

Reply:
or `--previous`... @jdlambert does `--rollback` ring odd for you too? or am i alone in that?

Reply:
I think we can purge the versions json once if we have any breaking changes and we don't want to support versions older than a specific one.

In my case I wanted to maintain all the configuration of my bond and only test the FW upgrade from one specific version to the latest one, I don't think we can do that in any other way...

Reply:
Now that you point it out, yeah, "rollback" has that connotation for me as well.

Reply:
what about `--version-num`? or `--version-index`. `--previous 0` or `--oldversion 0` sound weird for me

Reply:
How about `--age`?